src/borg/archive.py

```diff
@@ -469,10 +476,17 @@ def save(self, name=None, comment=None, timestamp=None, additional_metadata=None
         metadata.update(additional_metadata or {})
         metadata = ArchiveItem(metadata)
         data = self.key.pack_and_authenticate_metadata(metadata.as_dict(), context=b'archive')
+        return data
+
+    def save(self, name=None, comment=None, timestamp=None, additional_metadata=None):
+        name = self._check_name(name)
+        self.items_buffer.flush(flush=True)
+        data = self.pack_archive_item(name, comment, timestamp, additional_metadata)
```

Comment:
hmm, so _check_name is called twice (once by pack_archive_item)?

if intended, maybe add a comment about why.